### PR TITLE
Disable Select migration network when needed

### PR DESCRIPTION
**Pull request by gildub**
- Lift up `commonNetworkAdapters` logic to `VMwareProviderHostsTable` to enable/disable "select migration network" button if there are or not common adapters between hosts.
- Display a Tooltip messages when no selection has been made:
select-message1
- Display a Tooltip message when selected hosts offers no adapters in common:
select-message2

Resolves https://github.com/konveyor/virt-ui/issues/261

### CODE REVIEW

**Comment by gildub** [comment]
@vconzola, I've used initial text for the tooltips, please let us know if we require more specific messages.

**Review by mturley — DISMISSED** [review]
LGTM pending @vconzola 's review

**Comment by vconzola** [comment]
LGTM.

File: src/app/Providers/components/VMwareProviderHostsTable/VMwareProviderHostsTable.tsx
Comment (mturley) [inline src/app/Providers/components/VMwareProviderHostsTable/VMwareProviderHostsTable.tsx:72]: Sorry to nitpick after my prior approval @gildub , but I realized this could be simpler. Remember that each time the selected items change, the whole component re-renders and these functions are reinitialized. They are always called with the same values and will return the same thing in all calls of a given render. So they don't need to be functions. It could just be one variable:

```ts
const commonNetworkAdapters =
  selectedItems.length > 0
    ? selectedItems[0].networkAdapters.filter(({ ipAddress }) =>
        selectedItems.every((host) => host.networkAdapters.some((na) => na.ipAddress === ipAddress))
      )
    : [];
```

And then below instead of `!hasCommonNetworkAdapters()` you could use `commonNetworkAdapters.length === 0`, and instead of passing `commonNetworkAdapters(selectedItems)` you can just pass `commonNetworkAdapters`.

It's simpler, and it has the benefit of only running the filter function once per render (currently it'll run 3 times).

**Comment by mturley** [comment]
CI results are messed up here because of the GH Actions outage, but should fix themselves when you push another commit

File: src/app/Providers/components/VMwareProviderHostsTable/VMwareProviderHostsTable.tsx
Comment (gildub) [reply src/app/Providers/components/VMwareProviderHostsTable/VMwareProviderHostsTable.tsx:72]: That's pretty slick!

**Review by mturley — APPROVED** [review]

**Pull request merged**
